### PR TITLE
Output sourcemap file if options given in lessOptions

### PR DESCRIPTION
**Pull request by joefiorini**
This will output a sourceMap file using either the name of the main output file, or the filename from `sourceMapURL` if given.

### CODE REVIEW

**Comment by joefiorini** [comment]
@gabrielgrant assuming you accept this, could you also please publish `0.2.0` to NPM so it's easy to pull into my project? Thanks!

File: index.js [hunk 1]
```diff
@@ -43,13 +56,19 @@ LessCompiler.prototype.updateCache = function (srcDir, destDir) {
...
+
+	    return resolve(output);
+	  }
+
+    fs.writeFile(destFile, output.css, { encoding: 'utf8' }, fileCallback);
 
-          return resolve(output);
-        });
+	  if (lessOptions.hasOwnProperty('sourceMap')) {
+	    writeSourcemapFile(destFile, output.map, lessOptions.sourceMap, fileCallback);
```
Comment (jasonmit) [inline index.js]: We are resolving the same promise twice.  Might be better to use push these async methods to some promise array and use RSVP.all to resolve the containing Promise.

File: index.js [hunk 2]
```diff
@@ -43,13 +56,19 @@ LessCompiler.prototype.updateCache = function (srcDir, destDir) {
...
-          }
+	  function fileCallback (err) {
+	    if (err) {
+	      return reject(err);
+	    }
+
+	    return resolve(output);
+	  }
+
+    fs.writeFile(destFile, output.css, { encoding: 'utf8' }, fileCallback);
```
Comment (jasonmit) [inline index.js]: Tabbing is off a bit here

File: index.js [hunk 3]
```diff
@@ -11,6 +11,19 @@ module.exports = LessCompiler;
 LessCompiler.prototype = Object.create(CachingWriter.prototype)
 LessCompiler.prototype.constructor = LessCompiler
 
+function writeSourcemapFile(destFile, mapContent, mapOptions, callback) {
+  var filename = destFile.replace('.css', '.map');
+
+  if (mapOptions.hasOwnProperty('sourceMapURL')) {
+    outputDir = path.dirname(destFile);
```
Comment (jasonmit) [inline index.js]: Missing var

File: index.js (hunk 1 above)
Comment (joefiorini) [inline index.js]: Good call. I'll revamp this.

File: index.js (hunk 3 above)
Comment (joefiorini) [inline index.js]: Wow, that's terrible. I'll fix that.

**Comment by jasonmit** [comment]
Ping @joefiorini

**Comment by joefiorini** [comment]
If I don't get to this this weekend, I'll work on it Monday. I'd like to get it wrapped up as well.

**Comment by jasonmit** [comment]
:+1:

**Comment by jasonmit** [comment]
Closing since it's been inactive.  Please submit once it's finalized :+1:

**Pull request closed without merging**